src/napari/_qt/layer_controls/widgets/qt_contrast_limits.py

```diff
@@ -0,0 +1,261 @@
+        )
+
+        # Setup widgets
+        self.auto_scale_bar = AutoScaleButtons(layer, parent)
+        self.auto_scale_bar_label = QtWrappedLabel(trans._('auto-contrast:'))
+        self.contrast_limits_slider = _QDoubleRangeSlider(
+            Qt.Orientation.Horizontal,
+        )
+        self.contrast_limits_slider.show_clim_popup.connect(
+            self.show_clim_popup
+        )
+        decimals = range_to_decimals(
+            self._layer.contrast_limits_range, self._layer.dtype
+        )
+        self.contrast_limits_slider.setRange(
+            *self._layer.contrast_limits_range
+        )
+        self.contrast_limits_slider.setSingleStep(10**-decimals)
+        self.contrast_limits_slider.setValue(self._layer.contrast_limits)
+        self.contrast_limits_slider.setToolTip(
+            trans._('Right click for detailed slider popup.')
+        )
+
+        self.clim_popup = None
+
+        connect_setattr(
+            self.contrast_limits_slider.valueChanged,
+            self._layer,
+            'contrast_limits',
+        )
+        connect_setattr(
+            self.contrast_limits_slider.rangeChanged,
+            self._layer,
+            'contrast_limits_range',
+        )
+
+        self.contrast_limits_slider_label = QtWrappedLabel(
+            trans._('contrast limits:')
+        )
+
+    def show_clim_popup(self):
+        self.clim_popup = QContrastLimitsPopup(self._layer, self.parent())
+        self.clim_popup.setParent(self.parent())
+        self.clim_popup.move_to('top', min_length=650)
+        self.clim_popup.show()
+
+    def _on_contrast_limits_change(self):
+        """Receive layer model contrast limits change event and update slider."""
+        with qt_signals_blocked(self.contrast_limits_slider):
+            self.contrast_limits_slider.setValue(self._layer.contrast_limits)
+
+        if self.clim_popup:
+            with qt_signals_blocked(self.clim_popup.slider):
+                self.clim_popup.slider.setValue(self._layer.contrast_limits)
+
+    def _on_contrast_limits_range_change(self):
+        """Receive layer model contrast limits change event and update slider."""
+        with qt_signals_blocked(self.contrast_limits_slider):
+            decimals = range_to_decimals(
+                self._layer.contrast_limits_range, self._layer.dtype
+            )
+            self.contrast_limits_slider.setRange(
+                *self._layer.contrast_limits_range
+            )
+            self.contrast_limits_slider.setSingleStep(10**-decimals)
+
+        if self.clim_popup:
+            with qt_signals_blocked(self.clim_popup.slider):
+                self.clim_popup.slider.setRange(
+                    *self._layer.contrast_limits_range
+                )
+
+    def get_widget_controls(self) -> list[tuple[QtWrappedLabel, QWidget]]:
+        return [
+            (self.contrast_limits_slider_label, self.contrast_limits_slider),
+            (self.auto_scale_bar, self.auto_scale_bar_label),
```

Comment:
```suggestion
            (self.auto_scale_bar_label, self.auto_scale_bar),
```

<img width="278" height="89" alt="Image" src="https://github.com/user-attachments/assets/3a1869ec-9241-498a-a60d-1f2484111869" />